superset-frontend/src/explore/components/controls/ContourControl/ContourPopoverControl.tsx

```diff
@@ -0,0 +1,311 @@
+      'g' in contour.color &&
+      typeof contour.color.g === 'number' &&
+      'b' in contour.color &&
+      typeof contour.color.b === 'number' &&
+      'a' in contour.color &&
+      typeof contour.color.a === 'number';
+
+    const errors = determineErrorMap(contour);
+    setValidationErrors(errors);
+
+    setIsComplete(
+      isIsoband
+        ? validLower && validUpper && validColor
+        : validLower && validColor && validStrokeWidth,
+    );
+  }, [contour, currentTab]);
+
+  const onTabChange = (activeKey: any) => {
+    setCurrentTab(activeKey);
+  };
+
+  const updateStrokeWidth = (value: number | string) => {
+    const newContour = { ...contour };
+    newContour.strokeWidth = value;
+    setContour(newContour);
+  };
+
+  const updateColor = (rgb: colorType) => {
+    const newContour = { ...contour };
+    newContour.color = { ...rgb, a: 100 };
+    setContour(newContour);
+  };
+
+  const updateLowerThreshold = (value: number | string) => {
+    const newContour = { ...contour };
+    newContour.lowerThreshold = value;
+    setContour(newContour);
+  };
+
+  const updateUpperThreshold = (value: number | string) => {
+    const newContour = { ...contour };
+    newContour.upperThreshold = value;
+    setContour(newContour);
+  };
+
+  const containsErrors = () => {
+    const keys = Object.keys(validationErrors);
+    return keys.some(key => validationErrors[key].length > 0);
+  };
+
+  const handleSave = () => {
+    if (isComplete && onSave) {
+      onSave(convertContourToNumeric(contour));
+    }
+  };
+
+  const isobandSection = () => (
+    <div className="isoline-popover-section">
+      <StyledRow>
+        <Col flex="1">
+          <ControlHeader
+            name="isoband-threshold-lower"
+            label="Lower Threshold"
+            description="The lower limit of the threshold range of the Isoband"
+            validationErrors={validationErrors.lowerThreshold}
+            hovered
+          />
+          <TextControl
+            value={contour.lowerThreshold}
+            onChange={updateLowerThreshold}
+          />
+        </Col>
+        <Col flex="1">
+          <ControlHeader
+            name="isoband-threshold-upper"
+            label="Upper Threshold"
+            description="The upper limit of the threshold range of the Isoband"
+            validationErrors={validationErrors.upperThreshold}
+            hovered
+          />
+          <TextControl
+            value={contour.upperThreshold}
+            onChange={updateUpperThreshold}
+          />
+        </Col>
+      </StyledRow>
+      <StyledRow>
+        <Col flex="1">
+          <ControlHeader
+            name="isoband-color"
+            label="Color"
+            description="The color of the isoband"
+            validationErrors={validationErrors.color}
+            hovered
+          />
+          <ColorPickerControl
+            value={typeof contour === 'object' && contour?.color}
```

Comment:
In other components we're not checking for contour's type so I'm assuming it's not necessary here?